src/timer/timer_game_economy.cpp

```diff
@@ -38,6 +38,41 @@ TimerGameEconomy::Month TimerGameEconomy::month = {};
 TimerGameEconomy::Date TimerGameEconomy::date = {};
 TimerGameEconomy::DateFract TimerGameEconomy::date_fract = {};
 
+/**
+ * Converts a Date to a Year, Month & Day.
+ * @param date the date to convert from
+ * @param ymd  the year, month and day to write to
+ */
+/* static */ void TimerGameEconomy::ConvertDateToYMD(TimerGameEconomy::Date date, TimerGameEconomy::YearMonthDay *ymd)
+{
+	/* If we're using real-time units, economy months have 30 days and an economy year has 360 days.
+	 * Otherwise, we need to keep economy months and years in sync with the calendar. */
+	if (::_settings_game.economy.use_realtime_units) {
+		ymd->year = TimerGameEconomy::date.base() / EconomyTime::DAYS_IN_ECONOMY_YEAR;
+		ymd->month = TimerGameEconomy::date.base() % EconomyTime::DAYS_IN_ECONOMY_YEAR / EconomyTime::DAYS_IN_ECONOMY_MONTH;
+		ymd->day = TimerGameEconomy::date.base() % EconomyTime::DAYS_IN_ECONOMY_MONTH;
+	} else {
+		CalendarConvertDateToYMD(date, ymd);
+	}
+}
+
+/**
+* Converts a Date to a Year, Month & Day.
+* @param date the date to convert from
+* @param ymd  the year, month and day to write to
+*/
+/* static */ TimerGameEconomy::Date TimerGameEconomy::ConvertYMDToDate(TimerGameEconomy::Year year, TimerGameEconomy::Month month, TimerGameEconomy::Day day)
+{
+	/* If we're using real-time units, economy months have 30 days and an economy year has 360 days.
+	 * Otherwise, we need to keep economy months and years in sync with the calendar. */
+	if (::_settings_game.economy.use_realtime_units) {
+		const int total_months = (year.base() * EconomyTime::MONTHS_IN_YEAR) + month;
+		return (total_months * EconomyTime::DAYS_IN_ECONOMY_MONTH) + day - 1; // Day is 1-indexed and must be converted to 0-indexed.
```

Comment:
"and must be" .. why? If you make a comment like that, at least say the way. The "must be" feels like someone has a gun to your head :P

Suggestion: `Day is 1-indexed, hence the "- 1"`

Or something.